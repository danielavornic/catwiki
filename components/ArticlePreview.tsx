export const ArticlePreview = () => (
  <div className="flex flex-col md:flex-row md:items-center">
    <div className="mb-14 md:mb-0 md:mr-10 lg:mr-20 lg:max-w-[600px] 2xl:max-w-[700px]">
      <h2 className="text-3xl lg:text-4xl xl:text-5xl text-amber-900 mb-4">
        Why should you <br /> have a cat?
      </h2>
      <div className="w-12 h-[4px] bg-amber-900 mb-10"></div>
      <p className="lg:text-lg xl:text-xl">
        Having a cat around you can actually trigger the release of calming chemicals in your body
        which lower your stress and anxiety levels.
        <br />
        <br />
        Owning a cat can bring unconditional love and companionship to your life. Having a feline
        friend can also help to relieve stress and improve your heart health. Owning a cat can be an
        extremely rewarding relationship.
      </p>
    </div>
    <div className="columns-2 gap-4">
      <img src="/images/image-2.png" className="w-full mb-4" alt="Cat" />
      <img src="/images/image-3.png" className="w-[70%] float-right" alt="Cat" />
      <img src="/images/image-1.png" className="w-full float-right" alt="Cat" />
    </div>
  </div>
);
