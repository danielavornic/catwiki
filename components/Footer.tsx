import Link from 'next/link';

export const Footer = () => {
  return (
    <footer className="container mx-auto px-4 w-full md:px-2 mt-10 lg:mt-14 xl:mt-[100px] 2xl:mt[120px]">
      <div className="bg-black text-white rounded-t-[36px] pt-8 pb-6 px-6 md:px-12">
        <div className="flex flex-col sm:flex-row sm:justify-between sm:items-center">
          <Link href="/">
            <a>
              <img src="/images/catwiki-logo-white.svg" width="80px" />
            </a>
          </Link>
          <p className="text-sm mt-4 sm:mt-0">
            Â© created by <b>Daniela Vornic</b>
          </p>
        </div>
      </div>
    </footer>
  );
};
