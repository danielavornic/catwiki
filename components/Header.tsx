import Link from 'next/link';

export const Header = () => {
  return (
    <header className="container mx-auto p-4 mb-4 md:px-2">
      <Link href="/">
        <a>
          <img src="/images/catwiki-logo.svg" />
        </a>
      </Link>
    </header>
  );
};
