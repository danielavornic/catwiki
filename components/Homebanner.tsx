import { SearchInput } from './SearchInput';
import { SearchedBreeds } from './SearchedBreeds';

const imgBgStyles = [
  "bg-[url('/images/hero-image-sm.png')]",
  "xs:bg-[url('/images/hero-image-md.png')]",
  "lg:bg-[url('/images/hero-image-lg.png')]",
  'min-h-[160px]',
  'xs:min-h-[200px]',
  'sm:min-h-[300px]',
  'md:min-h-[420px]',
  'lg:min-h-[500px]',
  'xl:min-h-[600px]',
  'rounded-t-[36px]',
  'p-6',
  'sm:p-8',
  'md:p-12',
  'lg:p-20',
  'bg-cover',
  'bg-no-repeat',
  'bg-right',
  'text-white',
  'flex',
  'items-center',
].join(' ');

const textStyles = [
  'text-xs',
  'sm:text-sm',
  'md:text-lg',
  'lg:text-2xl',
  'max-w-[130px]',
  'sm:max-w-[180px]',
  'md:max-w-[200px]',
  'lg:max-w-[370px]',
  'mt-2',
  'md:mt-4',
  'mb-4',
  'md:mb-8',
  'lg:mb-10',
].join(' ');

export const Homebanner = () => {
  return (
    <div className="container mx-auto px-4 md:px-2">
      <div className={imgBgStyles}>
        <div>
          <img
            src="/images/catwiki-logo-white.svg"
            className="w-[100px] sm:w-[200px] lg:w-[300px]"
          />
          <h1 className={textStyles}>Get to know more about your cat breed</h1>
          <SearchInput />
        </div>
      </div>
      <SearchedBreeds />
    </div>
  );
};
