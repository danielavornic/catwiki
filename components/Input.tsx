import classNames from 'classnames';

type InputSize = 'medium' | 'small' | 'large';

interface InputProps extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'size'> {
  size?: InputSize;
}

export const Input = ({ className, size = 'medium', type, ...props }: InputProps) => {
  return (
    <input
      className={classNames(className, 'rounded transition', {
        'py-2 px-4': size === 'medium',
        'bg-blue-500 hover:bg-blue-600 cursor-pointer text-white': type === 'submit',
        'border focus:outline-none hover:border-gray-500 focus:border-gray-500': type !== 'submit',
      })}
      type={type}
      {...props}
    />
  );
};
