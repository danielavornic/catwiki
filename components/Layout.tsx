import Head from 'next/head';
import { PropsWithChildren } from 'react';

import { Footer, Header } from '.';

interface LayoutProps {
  title?: string;
  description?: string;
}

export const Layout = ({
  title = 'Catwiki',
  description = 'Explore kittens with CatWiki',
  children,
}: PropsWithChildren<LayoutProps>) => {
  return (
    <>
      <Head>
        <title>{title}</title>
        <meta name="description" content={description} />
      </Head>

      <Header />
      <main className="min-h-[80vh] sm:min-h-[85vh]">{children}</main>
      <Footer />
    </>
  );
};
