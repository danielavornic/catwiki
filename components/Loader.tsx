import { MutatingDots } from 'react-loader-spinner';

export const Loader = () => (
  <MutatingDots
    height="100"
    width="100"
    color="#291507"
    secondaryColor="#291507"
    radius="12.5"
    ariaLabel="mutating-dots-loading"
    wrapperStyle={{
      width: '100vw',
      height: '100vh',
      alignItems: 'center',
      justifyContent: 'center',
    }}
    visible={true}
  />
);
