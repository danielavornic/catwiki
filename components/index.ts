export * from './Header';
export * from './Footer';
export * from './Layout';
export * from './Homebanner';
export * from './SearchInput';
export * from './SearchedBreeds';
