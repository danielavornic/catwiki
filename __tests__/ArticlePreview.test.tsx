import { ArticlePreview } from '@/components/ArticlePreview';
const ReactTestRenderer = require('react-test-renderer');

describe('Article Preview', () => {
  it('renders correctly', () => {
    const tree = ReactTestRenderer.create(<ArticlePreview />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
