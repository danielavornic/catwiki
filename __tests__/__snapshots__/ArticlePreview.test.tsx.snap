// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Preview renders correctly 1`] = `
<div
  className="flex flex-col md:flex-row md:items-center"
>
  <div
    className="mb-14 md:mb-0 md:mr-10 lg:mr-20 lg:max-w-[600px] 2xl:max-w-[700px]"
  >
    <h2
      className="text-3xl lg:text-4xl xl:text-5xl text-brown-500 mb-4"
    >
      Why should you 
      <br />
       have a cat?
    </h2>
    <div
      className="w-12 h-[4px] bg-brown-500 mb-10"
    />
    <p
      className="lg:text-lg xl:text-xl"
    >
      Having a cat around you can actually trigger the release of calming chemicals in your body which lower your stress and anxiety levels.
      <br />
      <br />
      Owning a cat can bring unconditional love and companionship to your life. Having a feline friend can also help to relieve stress and improve your heart health. Owning a cat can be an extremely rewarding relationship.
    </p>
  </div>
  <div
    className="columns-2 gap-4"
  >
    <img
      alt="Cat"
      className="w-full mb-4"
      src="/images/image-2.png"
    />
    <img
      alt="Cat"
      className="w-[70%] float-right"
      src="/images/image-3.png"
    />
    <img
      alt="Cat"
      className="w-full float-right"
      src="/images/image-1.png"
    />
  </div>
</div>
`;
